offer-manager-domain/src/main/groovy/life/qbic/business/customers/update/UpdateCustomer.groovy

```diff
@@ -1,10 +1,58 @@
 package life.qbic.business.customers.update
 
+import life.qbic.datamodel.dtos.business.Customer
+
+import life.qbic.business.exceptions.DatabaseQueryException
+
 /**
- * A project manager wants to modify the customer information
+ * This use case updates an existing customer in the system. New Affiliations of the customer are added to the respective table.
+ * If other changes are made to the customer, a new customer is created in the system and the old customer is set to inactive.
  *
  * @since: 1.0.0
- * @author: Tobias Koch
  */
-class UpdateCustomer {
+class UpdateCustomer implements UpdateCustomerInput {
+
+  private UpdateCustomerDataSource dataSource
+  private UpdateCustomerOutput output
+
+  UpdateCustomer(UpdateCustomerOutput output, UpdateCustomerDataSource dataSource){
+    this.output = output
+    this.dataSource = dataSource
+  }
+  
+  // determines if customer properties other than affiliations have changed
+  private boolean hasBasicCustomerDataChanged(Customer existingCustomer, Customer newCustomer) {
+    boolean noFundamentalChange = existingCustomer.firstName.equals(newCustomer.firstName)
+    && existingCustomer.lastName.equals(newCustomer.lastName)
+    && existingCustomer.emailAddress.equals(newCustomer.emailAddress)
+    && existingCustomer.title.equals(newCustomer.title)
+    
+    return !noFundamentalChange
+  }
+
+  @Override
+  void updateCustomer(String customerId, Customer customer) {
+    Customer existingCustomer = dataSource.getCustomer(Integer.parseInt(customerId))
+    boolean customerChanged = hasBasicCustomerDataChanged(existingCustomer, customer)
+    try {
+      if(customerChanged) {
+        dataSource.updateCustomer(customerId, customer)
+      } else {
+        dataSource.updateCustomerAffiliations(customerId, customer.affiliations)
+      }
+      try {
+        output.customerUpdated(customer)
+      } catch (Exception ignored) {
+        //quiet output message failed
+      }
```

Comment:
Why do you nest it here too? Is it not sufficient to only have one try-catch block?

Reply:
The other use case made a difference between failing to perform the action, which is reported to the customer, and failing at returning the output of the action, which is silent. I copied it, not sure if it makes sense.

Reply:
I think it's not necessary and I would remove it to keep it simple

Reply:
ok, will change it